### PR TITLE
Removes dependency on internal version of resource

### PR DESCRIPTION
**Pull request by seans3**
* Removes code dependency on internal version of Pod
* Removes unused code

1) [Umbrella Issue: Remove kubectl dependencies on kubernetes/kubernetes](https://github.com/kubernetes/kubectl/issues/80)
2) [Remove Kubectl dependencies on kubernetes/pkg/api and kubernetes/pkg/apis](https://github.com/kubernetes/kubectl/issues/83)

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/conditions.go [hunk 1]
```diff
@@ -117,38 +79,21 @@ func PodRunningAndReady(event watch.Event) (bool, error) {
...
 	}
 	switch t := event.Object.(type) {
-	case *api.Pod:
-		switch t.Status.Phase {
-		case api.PodFailed, api.PodSucceeded:
-			return false, ErrPodCompleted
-		case api.PodRunning:
-			return pod.IsPodReady(t), nil
-		}
 	case *corev1.Pod:
```
Comment (apelisse) [inline pkg/kubectl/conditions.go]: How do you know it's never going to be a `api.Pod`?

File: pkg/kubectl/conditions.go [hunk 2]
```diff
@@ -117,38 +79,21 @@ func PodRunningAndReady(event watch.Event) (bool, error) {
...
 		case corev1.PodRunning:
-			return podv1.IsPodReady(t), nil
-		}
-	}
-	return false, nil
-}
-
-// PodNotPending returns true if the pod has left the pending state, false if it has not,
-// or an error in any other case (such as if the pod was deleted).
-func PodNotPending(event watch.Event) (bool, error) {
```
Comment (apelisse) [inline pkg/kubectl/conditions.go]: Removing dead code, yay!

File: pkg/kubectl/conditions.go (hunk 1 above)
Comment (seans3) [reply pkg/kubectl/conditions.go]: Both "PodCompleted" and "PodRunningAndReady" functions are only called once each in run.go. In fact, they are only called as an argument in this file: "waitForPod".  Within "waitForPod", the functions are called, and only external versions of Pod are used.

File: pkg/kubectl/conditions.go (hunk 2 above)
Comment (seans3) [reply pkg/kubectl/conditions.go]: :+1:

**Pull request merged**